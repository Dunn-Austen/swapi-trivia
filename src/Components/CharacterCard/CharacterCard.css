.character-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

.character-card {
    align-items: center;
    background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url("../../Images/Vader.jpg");
    background-blend-mode: saturation;
    background-size: 100% 100%;
    background-color: #5C5E5F;
    background-blend-mode: screen, difference, darken;
    box-shadow: 8px 8px 30px white;
    display: flex;
    flex-direction: column;
    height: 175px;
    margin: 20px;
    padding: 20px;
    justify-content: space-around;
    width: 165px;
  }
  .character-card:hover {
    box-shadow: 0 0 10px grey;
  }