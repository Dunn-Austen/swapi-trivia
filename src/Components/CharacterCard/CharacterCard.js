import React from 'react';
import './CharacterCard.css';
import PropTypes from 'prop-types';

const CharacterCard = ({name, homeworld, population, species, urelatedfilms}) => {
console.log(name)
  return (
    <article className="character-card">
      <h2 className="character-name">{name}</h2>
      <h2 className="character-homeworld">{homeworld}</h2>
      <h2 className="character-population">{population}</h2>
      <h2 className="character-species">{species}</h2>
      {/* <h2 className="character-relatedfilms">{relatedfilms}</h2> */}
      <button>View Characters</button>
</article>
  )
};

export default CharacterCard;

CharacterCard.propTypes = {
  selectedCharacters: PropTypes.array
}
