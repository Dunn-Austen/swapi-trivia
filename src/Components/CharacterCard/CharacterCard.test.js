import React from 'react'
import { shallow } from 'enzyme'

import CharacterCard from '../CharacterCard/CharacterCard';


describe('CharacterCard', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CharacterCard
            name="Foster"
            films="Major League"
            homeworld="Big Apple"
            population="55555"
            species="Alien"
        />)
    })

    it('should match the snapshot with all data passed correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

});
