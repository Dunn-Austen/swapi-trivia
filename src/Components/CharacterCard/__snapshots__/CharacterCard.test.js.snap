// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match the snapshot with all data passed correctly 1`] = `
<article
  className="character-card"
>
  <h2
    className="character-name"
  >
    Foster
  </h2>
  <h2
    className="character-homeworld"
  >
    World: 
    Big Apple
  </h2>
  <h2
    className="character-population"
  >
    Pop: 
    55555
  </h2>
  <h2
    className="character-species"
  >
    Species: 
    Alien
  </h2>
  <button
    className="fave-btn"
    onClick={[Function]}
  >
    Fave
  </button>
</article>
`;
